Client/Screen.py

```diff
@@ -171,73 +153,102 @@ def onreturn_play_menu_input_ip(self, val):  # check onreturn of playMenu IP inp
 
                 # acknowledged the connection error status, and now set back the flag to None
                 Client.Client.get_instance().connection_alive = None
-                self.is_conn_msg_shown = False
 
                 pg.time.wait(2500)
 
-        elif Client.Client.get_instance().connection_alive is not None:
-            self.connectingMenu.add_line('')
-
-            if Client.Client.get_instance().connection_alive:
-                self.connectingMenu.disable()
-                self.init_connecting_menu()
-                self.connectingMenu.add_line('Successfully connected.')
-            else:
-                self.connectingMenu.add_line('Connection error, please try again!')
-
-            self.is_conn_msg_shown = True
-
-    def connecting_menu_player_count_selector_onchange(self, txt_value, value):
-        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.CHANGE_PLAYER_NUMBER,
-                                          target=climess.Target.GAME)
+    def _connection_menu_player_count_selector_onchange(self, _, value):
+        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.CHANGE_PLAYER_NUMBER, target=climess.Target.GAME)
         msg.new_number = value
         Client.Client.get_instance().send_message(msg)
 
-    def connecting_menu_start_pressed(self):
-        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.START_GAME_MANUALLY,
-                                      target=climess.Target.GAME)
+    def _connection_menu_start_pressed(self):
+        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.START_GAME_MANUALLY, target=climess.Target.GAME)
         Client.Client.get_instance().send_message(msg)
 
-    def game_screen(self):
-        self.screen.fill((0, 0, 0))  # black bg
+    #  More useful functions below!!!
+    def update(self, events):
+        # draw the adequate screen (according to the state)
+        if self.__screenState == sstatecons.ScreenState.MAIN_MENU:
+            if not self.__mainMenu.is_enabled():
+                self.__mainMenu.enable()
+            self.__mainMenu.mainloop(events)
+        elif self.__screenState == sstatecons.ScreenState.CONNECTION_MENU:
+            if not self.__connectionMenu.is_enabled():
+                self.__connectionMenu.enable()
+            self.__connectionMenu.mainloop(events)
+        elif self.__screenState == sstatecons.ScreenState.GAME:
+            self._game_screen()
+
+        running = True
+        for event in events:  # event handling - look at every event in the queue
+
+            if event.type == pg.KEYDOWN:  # did the user hit a key?
+
+                if event.key == pg.K_ESCAPE:  # Was it the Escape key? If so, stop the loop.
+                    running = False
+
+            # Did the user click the window close button? If so, stop the loop. w/o. doesn't work
+            if event.type == pg.QUIT:
+                running = False
```

Comment:
Outsourcing this into sg like a check_keyboard_exit_request function seems desirable for me :)